test/lib/sdkdriver/src/dmp.ts

```diff
@@ -0,0 +1,132 @@
+import { EventEmitter } from 'events';
+import Transport from './transport';
+
+/**
+ * DMP Request.
+ * @class
+ * @classdesc A {@link DMPRequest} contains the data associated with
+ *   the incoming request message along with a method for sending a
+ *   response.
+ */
+class DMPRequest {
+  private readonly _id: number;
+  private readonly _transport: Transport;
+  public readonly data: any;
+
+  /**
+   * Constructor.
+   * @param {number} id - Request ID
+   * @param {*} data - Request data
+   * @param {Transport} transport - The {@link Transport} used for
+   *   sending a response
+   */
+  constructor(id: number, data: any, transport: Transport) {
+    this._id = id;
+    this._transport = transport;
+    this.data = data;
+  }
+
+  /**
+   * Send a response over the underlying {@link Transport}.
+   * @param {*} data - Response data
+   * @returns {void}
+   */
+  sendResponse(data: any): void {
+    const { _id: id, _transport: transport } = this;
```

Comment:
If we're only accessing `_id` and `_transport` once, maybe the destructuring assignment doesn't buy us much?

Reply:
@markandrus I think we should use destructuring wherever it decreases verbosity.